compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -4897,6 +4897,14 @@ private BLangUnaryExpr createBLangUnaryExpr(Location location,
     }
 
     private BLangExpression createExpression(Node expression) {
+        if (expression.kind() == SyntaxKind.ASYNC_SEND_ACTION) {
+            // TODO: support async send as expression #29089
+            dlog.error(getPosition(expression), DiagnosticErrorCode.ASYNC_SEND_NOT_YET_SUPPORTED_AS_EXPRESSION);
+            Token missingIdentifier = NodeFactory.createMissingToken(SyntaxKind.IDENTIFIER_TOKEN,
+                    NodeFactory.createEmptyMinutiaeList(), NodeFactory.createEmptyMinutiaeList());
+            return (BLangExpression) createActionOrExpression(missingIdentifier);
```

Comment:
should be wrap `missingIdentifier` wth NodeFactory.createSimpleNameReferenceNode() before passing to `createActionOrExpression()`?
Also, I think we can have a single return within the method in following way.
```suggestion
            expression = NodeFactory.createSimpleNameReferenceNode(missingIdentifier);
```